source/_patterns/01-atoms/button/_button.twig

```diff
@@ -0,0 +1,32 @@
+{#
+  Button!
+
+  button_element: String: button|a|input choose the element type for your button
+  button_type: String: the html 'type' attribute
+  button_outline: Boolean: True|False, remove all background colors and images from a button
+  button_color: String: primary|secondary|success|danger|warning|info|light|dark
+  button_size: String: lg|sm set whether the button is large or small
+  button_block:  Boolean: True|False set button to occupy the full width the parent element
+  button_active: Boolean: True|False set button to appear pressed
+  button_toggle: Boolean: True|False set button to be toggleable
+  button_link: String: the URL the button will lead to
+  button_value: String: the html 'value' attribute
+  button_disabled: Boolean: True|False set button to appear disabled
+  button_text: String: Display text on the button
+
+  See https://getbootstrap.com/docs/4.0/components/buttons/ for more details
+#}
+
+<{{ button_element|default('button') }}
+type="{{ button_type }}"
+class="btn btn-{% if button_outline == TRUE  %}outline-{% endif %}{{ button_color|default('primary') }}
+  {% if button_size %} btn-{{ button_size }}{% endif %}
```

Comment:
I'd like you to take a look at recent work on projects for how we can load up these attributes as a concatenated string in a `{% set %}` block. This vastly simplifies this markup.